### ISSUE TITLE
Significant accuracy drop during conversion of DistillBert.

### ISSUE THREAD

**Issue by ichernob** [tensorflow/tensorflow#43476]
**System information**
- OS: Ubuntu 18.04.5 LTS (x86_64)
- TensorFlow installed from (source or binary):
- TensorFlow version (or github SHA if from source):

**Command used to run the converter or code if you’re using the Python API**

```
converter = tf.lite.TFLiteConverter.from_keras_model(tf_model)
# For normal conversion:
#converter.experimental_new_converter = True
#converter.optimizations = [tf.lite.Optimize.DEFAULT]
converter.target_spec.supported_ops = [tf.lite.OpsSet.SELECT_TF_OPS]
tflite_model = converter.convert()
```

**The output from the converter invocation**

```
WARNING:tensorflow:Skipping full serialization of Keras layer <tensorflow.python.keras.layers.core.Dropout object at 0x7f6cc9b669e8>, because it is not built.
WARNING:tensorflow:Skipping full serialization of Keras layer <tensorflow.python.keras.layers.core.Dropout object at 0x7f6c426cba58>, because it is not built.
WARNING:tensorflow:Skipping full serialization of Keras layer <tensorflow.python.keras.layers.core.Dropout object at 0x7f6c3c4fd080>, because it is not built.
WARNING:tensorflow:Skipping full serialization of Keras layer <tensorflow.python.keras.layers.core.Dropout object at 0x7f6c3c50d668>, because it is not built.
WARNING:tensorflow:Skipping full serialization of Keras layer <tensorflow.python.keras.layers.core.Dropout object at 0x7f6c3c519c50>, because it is not built.
WARNING:tensorflow:Skipping full serialization of Keras layer <tensorflow.python.keras.layers.core.Dropout object at 0x7f6c3c4b5278>, because it is not built.
WARNING:tensorflow:From /usr/local/lib/python3.6/dist-packages/tensorflow/python/training/tracking/tracking.py:111: Model.state_updates (from tensorflow.python.keras.engine.training) is deprecated and will be removed in a future version.
Instructions for updating:
This property should not be used in TensorFlow 2.0, as updates are applied automatically.
WARNING:tensorflow:From /usr/local/lib/python3.6/dist-packages/tensorflow/python/training/tracking/tracking.py:111: Layer.updates (from tensorflow.python.keras.engine.base_layer) is deprecated and will be removed in a future version.
Instructions for updating:
This property should not be used in TensorFlow 2.0, as updates are applied automatically.
INFO:tensorflow:Assets written to: /tmp/tmpsca4zbmo/assets
INFO:absl:Using experimental converter: If you encountered a problem please file a bug. You can opt-out by setting experimental_new_converter=False
```

**Also, please include a link to the saved model or GraphDef**

```
https://huggingface.co/distilbert-base-multilingual-cased#.
```

**Failure details**
- Producing wrong results and/or decrease in accuracy
The original model produces output:
```
TFBaseModelOutput([('last_hidden_state',
                    <tf.Tensor: shape=(1, 128, 768), dtype=float32, numpy=
                    array([[[ 0.07324436, -0.13601898,  0.24253392, ...,  0.2757071 ,
                              0.05075948, -0.067513  ],
                            [-0.00969718, -0.6601796 ,  0.07741702, ...,  0.20618156,
                             -0.3217669 ,  0.25172737],
                            [ 0.28169543, -0.36096776, -0.07832485, ...,  0.00399976,
                              0.07738406, -0.06696145],
                            ...,
                            [ 0.04910686, -0.37189117,  0.33814144, ...,  0.03981559,
                              0.14327443, -0.20109934],
                            [ 0.07308616, -0.35554424,  0.33334315, ...,  0.08324002,
                              0.1068868 , -0.22119546],
                            [ 0.06794482, -0.37025544,  0.3678054 , ...,  0.0837122 ,
                              0.11539332, -0.2085252 ]]], dtype=float32)>)])
```
The converted model gives:
```
array([[[ 0.19460055, -0.07871183,  0.348869  , ...,  0.56901807,
          0.40855035, -0.36336952],
        [ 0.111944  , -0.10565656,  0.47841206, ...,  0.7020322 ,
          0.1973463 , -0.56403255],
        [ 0.2462692 , -0.04639575,  0.47352695, ...,  0.6418205 ,
          0.26107264, -0.72952   ],
        ...,
        [ 0.2746247 , -0.06043699,  0.4579026 , ...,  0.7356582 ,
          0.4970783 , -0.6069904 ],
        [ 0.2640199 , -0.05569951,  0.4638841 , ...,  0.734777  ,
          0.48725146, -0.60665435],
        [ 0.25305653, -0.04610543,  0.44723004, ...,  0.7529757 ,
          0.47617406, -0.6056732 ]]], dtype=float32)
```

**Any other info / logs**

Source code to reproduce:

```
from transformers import TFDistilBertModel, DistilBertTokenizer, DistilBertConfig
import tensorflow as tf
from typing import Dict
import numpy as np

class myTFDistilBertModel(TFDistilBertModel):
    # DUMMY_INPUTS = [[7, 6, 0, 0, 1], [1, 2, 3, 0, 0], [0, 0, 0, 4, 5]]
    # tf.constant(np.zeros(shape=(1, 100), dtype=int).tolist())
    # For details refer https://github.com/huggingface/transformers/blob/7cbf0f722d23440f3342aafc27697b50ead5996b/src/transformers/modeling_tf_utils.py#L218
    def __init__(self, config, *inputs, **kwargs):
        super().__init__(config, *inputs, **kwargs)
    
    @property
    def dummy_inputs(self) -> Dict[str, tf.Tensor]:
        """
        Dummy inputs to build the network.
        Returns:
            :obj:`Dict[str, tf.Tensor]`: The dummy inputs.
        """
        return {"input_ids": tf.constant(np.zeros(shape=(1, 128), dtype=int).tolist())}

tf_model = myTFDistilBertModel.from_pretrained('distilbert-base-multilingual-cased', return_dict=True)
tf_tokenizer = DistilBertTokenizer.from_pretrained('distilbert-base-multilingual-cased', return_tensors='tf')

inputs = tf_tokenizer("le droit d'accès", padding='max_length', max_length=128, return_tensors='tf')
outputs  = tf_model(inputs)

input_spec = tf.TensorSpec([1, 128], tf.int32)
tf_model._set_inputs(input_spec, training=False)

converter = tf.lite.TFLiteConverter.from_keras_model(tf_model)
# For normal conversion:
#converter.experimental_new_converter = True
#converter.optimizations = [tf.lite.Optimize.DEFAULT]
converter.target_spec.supported_ops = [tf.lite.OpsSet.SELECT_TF_OPS]
tflite_model = converter.convert()

# Run the model with TensorFlow Lite
interpreter = tf.lite.Interpreter(model_content=tflite_model)
# Allocate tensors.
interpreter.allocate_tensors()

# Get input and output tensors.
input_details = interpreter.get_input_details()
output_details = interpreter.get_output_details()

input = tf_tokenizer("le droit d'accès", padding='max_length', max_length=128, return_tensors='tf')
interpreter.set_tensor(0, input['input_ids'])
interpreter.invoke()
interpreter_output = interpreter.get_tensor(output_details[0]["index"])
```

**Comment by Saduf2019** [comment]
@ruichernob 
I ran the code shared and face a different error, please find [gist here](https://colab.research.google.com/gist/Saduf2019/1a333e92acaf40bc14fccf6c966750fb/untitled419.ipynb), could you provide us with a colab gist replicating the issue faced.

**Comment by ruichernob** [comment]
@Saduf2019 As far as I understand you just need to '!pip install transformers'. Ther rest of the code works well.
I've changed shared notebook a little bit in order to let you observe the reported [issue](https://colab.research.google.com/gist/Saduf2019/1a333e92acaf40bc14fccf6c966750fb/untitled419.ipynb#scrollTo=BlFb98ps1y_8)

Just in case you're unable to fetch the changes:

Added transformers:
```
!pip install tf-nightly
!pip install transformers
```

The next cell didn't changed:
```
from transformers import TFDistilBertModel, DistilBertTokenizer, DistilBertConfig
import tensorflow as tf
from typing import Dict
import numpy as np

class myTFDistilBertModel(TFDistilBertModel):
    # DUMMY_INPUTS = [[7, 6, 0, 0, 1], [1, 2, 3, 0, 0], [0, 0, 0, 4, 5]]
    # tf.constant(np.zeros(shape=(1, 100), dtype=int).tolist())
    # For details refer https://github.com/huggingface/transformers/blob/7cbf0f722d23440f3342aafc27697b50ead5996b/src/transformers/modeling_tf_utils.py#L218
    def __init__(self, config, *inputs, **kwargs):
        super().__init__(config, *inputs, **kwargs)
    
    @property
    def dummy_inputs(self) -> Dict[str, tf.Tensor]:
        """
        Dummy inputs to build the network.
        Returns:
            :obj:`Dict[str, tf.Tensor]`: The dummy inputs.
        """
        return {"input_ids": tf.constant(np.zeros(shape=(1, 128), dtype=int).tolist())}

tf_model = myTFDistilBertModel.from_pretrained('distilbert-base-multilingual-cased', return_dict=True)
tf_tokenizer = DistilBertTokenizer.from_pretrained('distilbert-base-multilingual-cased', return_tensors='tf')

inputs = tf_tokenizer("le droit d'accès", padding='max_length', max_length=128, return_tensors='tf')
outputs  = tf_model(inputs)

input_spec = tf.TensorSpec([1, 128], tf.int32)
tf_model._set_inputs(input_spec, training=False)

converter = tf.lite.TFLiteConverter.from_keras_model(tf_model)
# For normal conversion:
#converter.experimental_new_converter = True
#converter.optimizations = [tf.lite.Optimize.DEFAULT]
converter.target_spec.supported_ops = [tf.lite.OpsSet.SELECT_TF_OPS]
tflite_model = converter.convert()

# Run the model with TensorFlow Lite
interpreter = tf.lite.Interpreter(model_content=tflite_model)
# Allocate tensors.
interpreter.allocate_tensors()

# Get input and output tensors.
input_details = interpreter.get_input_details()
output_details = interpreter.get_output_details()

input = tf_tokenizer("le droit d'accès", padding='max_length', max_length=128, return_tensors='tf')
interpreter.set_tensor(0, input['input_ids'])
interpreter.invoke()
interpreter_output = interpreter.get_tensor(output_details[0]["index"])
```

To observe interpreter output:
```
interpreter_output
```
That gives:
```
array([[[ 0.1946005 , -0.07871191,  0.34886912, ...,  0.56901795,
          0.4085501 , -0.36336952],
        [ 0.11194398, -0.1056565 ,  0.47841233, ...,  0.70203197,
          0.19734594, -0.56403244],
        [ 0.24626903, -0.0463957 ,  0.4735275 , ...,  0.64182055,
          0.2610727 , -0.72952026],
        ...,
        [ 0.2746247 , -0.06043694,  0.45790255, ...,  0.73565805,
          0.4970783 , -0.6069902 ],
        [ 0.26401982, -0.05569961,  0.4638846 , ...,  0.7347771 ,
          0.48725143, -0.60665435],
        [ 0.25305668, -0.0461056 ,  0.44722998, ...,  0.7529754 ,
          0.47617412, -0.6056729 ]]], dtype=float32)
```
To observe the model output:
```
outputs
```

That gives:
```
TFBaseModelOutput([('last_hidden_state',
                    <tf.Tensor: shape=(1, 128, 768), dtype=float32, numpy=
                    array([[[ 0.07324453, -0.13601895,  0.24253356, ...,  0.27570707,
                              0.05075945, -0.06751309],
                            [-0.00969684, -0.6601795 ,  0.07741669, ...,  0.20618166,
                             -0.32176685,  0.25172746],
                            [ 0.28169546, -0.36096776, -0.07832497, ...,  0.00399961,
                              0.07738377, -0.06696163],
                            ...,
                            [ 0.04910718, -0.37189072,  0.33814105, ...,  0.03981549,
                              0.14327444, -0.20109943],
                            [ 0.07308632, -0.35554397,  0.3333426 , ...,  0.08323996,
                              0.10688686, -0.22119588],
                            [ 0.06794469, -0.37025517,  0.36780512, ...,  0.08371204,
                              0.11539327, -0.20852552]]], dtype=float32)>)])
```

As we can observe, there is a significant mismatch.

**Comment by Saduf2019** [comment]
@ruichernob 
I ran the code shared and face a different issue, please find the [gist here](https://colab.research.google.com/gist/Saduf2019/d4b211faf3dc3e81c1caddba3952f413/untitled418.ipynb). Can you please share a colab gist with the error reported.

**Comment by ruichernob** [comment]
@Saduf2019 
I've been able to run your code without any errors. If you still faces the issue you can attempt comment
```
#input_spec = tf.TensorSpec([1, 128], tf.int32)
#tf_model._set_inputs(input_spec, training=False)
```

**Comment by karimnosseir** [comment]
Renjie, can you please have a look

Thanks

**Comment by tilakrayal** [comment]
@ichernob,
I tried to execute the provided code as mentioned and it failed with the different error. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/082b920884d8a9689af835da621f7bde/untitled726.ipynb) and confirm whether the issue got resolved on tf-nightly. Thank you!
